lib/trento/domain/sap_system/sap_system.ex

```diff
@@ -221,6 +224,35 @@ defmodule Trento.Domain.SapSystem do
     }
   end
 
+  # Deregister a database instance and emit a DatabaseInstanceDeregistered
```

Comment:
Something new that came to my mind.
We shouldn't send any event if the instance is not registered. This shouldn't happen if the commands are not launched from the cli though, but a good to have thing.
So we should have 2 extra conditions:
- SAP system is not even registered
- Instance is not in the registered SAP system

Anyway, we can work on that in a next PR.